internal/auth/options.go

```diff
@@ -326,13 +330,22 @@ func AssignProvider(opts *Options) func(*Authenticator) error {
 	}
 }
 
+// SetProvider is a function that takes an Options struct and assigns the
+// appropriate provider to the proxy.
+func SetProvider(provider providers.Provider) func(*Authenticator) error {
```

Comment:
I think the comment needs to be slightly altered with it's inputs

Reply:
Done